aospy_synthetic/test/test_db.py

```diff
@@ -3,78 +3,98 @@
 import unittest
 from subprocess import call
 
-from test_objs import (cases, models, example_projects, variables,
-                       regions)
-from aospy_synthetic.calc import Calc, CalcInterface
+from test_objs import (runs, models, projects, variables,
+                       regions, calc_objs)
 from aospy_synthetic.db.sqlalchemy.sqlalchemy_db import SQLAlchemyDB
-#from aospy_synthetic.SQLAlchemyDB import SQLBackend
+from aospy_synthetic.db.sqlalchemy.sqlalchemy_config import (ProjDB, ModelDB,
```

Comment:
Yes, I think it would be the case that every backend would need to implement DB objects at each object level.  That said I'm not completely sure how other backends might function -- if they are attribute driven (like SQLAlchemy) then I suppose one could make abstract versions of the DB objects that would simply enforce that they have certain attributes.